client/util.go

```diff
@@ -21,17 +22,15 @@ const docIDPrefixLength = docIDVersionLength + docIDNonceLength
 // pathnameToDocID encrypts a `pathname` to a document ID using `key`.
 func pathnameToDocID(pathname string, key [32]byte) (sserver1.DocumentID, error) {
 	var nonce [docIDNonceLength]byte
-	_, err := rand.Read(nonce[:])
-	if err != nil {
-		return sserver1.DocumentID(""), err
-	}
+	cksum := sha256.Sum256([]byte(pathname))
```

Comment:
Exactly. Sure I will do that. Generally speaking deterministic nonces and paddings are fine, as they are commonly used in the CBC mode block cipher construction, where they simply use counters for nonces (1, 2, 3...) In here we cannot do that so I guess we need to go with hashes.